libc/include/math.h.def

```diff
@@ -17,4 +17,6 @@
 
 %%public_api()
 
+#define isnan(x) __builtin_isnan(x)
```

Comment:
Should we put this macro into its own header? Do you have any suggestion for the name?

Reply:
If possible I'd like to leave this in `math-macros` but if it needs to be moved out then I'd say it should be defined in `isnan-macro.h` inside `include/llvm-libc-macros/`

Reply:
Do we expect every consumer of this header to have this builtin? We could possible do `if defined(__has_builtin) && __has_builtin(__builtin_isnan))` or omething.

Reply:
Does it mean that we still define `isnan` macro after defining all the `isnan` function?

If so then I think it makes sense to do the `TODO` in `math-macros.h`, that is to move `isinf`, `isfinite`, `isnan` macro definitions to `math-function-macros.h`, including it after `%%public_api` in here, and commenting about the reason we have to include it after all the function definitions.

Reply:
> Do we expect every consumer of this header to have this builtin?

Yes.  All versions of clang and gcc that we claim to support have this builtin. Therefor, we should not guard it with __has_builtin (same for other builtins).

Reply:
> that is to move isinf, isfinite, isnan

So isnan has isnan, isnanf, and isnanl.

isinf has isinf, isinff, isinfl,

finite has finite, finitef, finitel.  Note that finite is not isfinite{f|l}, so it lacks the symmetry that isinf and isnan have. Stated another way, isfinite (the type-generic macro) does not conflict with the finite, finitef, and finitel functions (like isnan and isinf do).

Reply:
For longer term maintainability I'd like to go with a design like what @lntue proposed where the primary math macros are included before `%%public_api` and the macros that might also be functions are defined in a separate header after.

Reply:
Done.